EIPS/eip-3074.md

```diff
@@ -56,39 +53,66 @@ The variable has the same scope as the program counter -- `authorized` persists
 
 ### `AUTH` (`0xf6`)
 
-A new opcode `AUTH` shall be created at `0xf6`. It shall take four stack element inputs and returns one stack element.
+A new opcode `AUTH` shall be created at `0xf6`. It shall take four stack element inputs (the last two describing a memory range), and it shall return one stack element.
 
 #### Input
 
+##### Stack
+
 | Stack      | Value        |
 | ---------- | ------------ |
 | `top - 0`  | `commit`     |
-| `top - 1`  | `yParity`    |
-| `top - 2`  | `r`          |
-| `top - 3`  | `s`          |
+| `top - 1`  | `authority`  |
+| `top - 2`  | `offset`     |
+| `top - 3`  | `length`     |
+
+##### Memory
+
+The final two stack arguments (`offset` and `length`) describe a range of memory. The format of the contents of that range is:
+
+ - `memory[offset    : offset+32]` - `yParity`
+ - `memory[offset+32 : offset+64]` - `r`
+ - `memory[offset+64 : offset+96]` - `s`
 
 #### Output
 
+##### Stack
+
 | Stack      | Value        |
 | ---------- | -------------|
 | `top - 0`  | `authorized` |
 
+##### Memory
+
+Memory is not modified by this instruction.
+
 #### Behavior
 
+If `length` is greater than 96, the extra bytes are ignored. Bytes outside the range (in the event `length` is less than 96) are treated as if they had been zeroes.
```

Comment:
This is only the case when we are using an ECDSA signature, right?

Reply:
Right, if we ever change the signature format, this paragraph would have to be changed.